sdk/loadtestservice/load-testing-rest/src/beginUploadAndValidateFile.ts

```diff
@@ -0,0 +1,190 @@
+      ) {
+        state.status = "running";
+      }
+
+      if (
+        fileValidationResponse.body.validationStatus === "VALIDATION_SUCCESS" ||
+        fileValidationResponse.body.validationStatus === "VALIDATION_NOT_REQUIRED"
+      ) {
+        state.status = "succeeded";
+        state.result = fileValidationResponse;
+      }
+
+      if (fileValidationResponse.body.validationStatus === "VALIDATION_FAILURE") {
+        state.status = "failed";
+        state.error = new Error(fileValidationResponse.body.validationStatus);
+      }
+
+      await processProgressCallbacks();
+
+      if (state.status === "failed") {
+        throw state.error;
+      }
+    },
+
+    pollUntilDone(pollOptions?: {
+      abortSignal?: AbortSignalLike;
+    }): Promise<TestGetFile200Response> {
+      return (resultPromise ??= (async () => {
+        const { abortSignal: inputAbortSignal } = pollOptions || {};
+        const { signal: abortSignal } = inputAbortSignal
+          ? new AbortController([inputAbortSignal, abortController.signal])
+          : abortController;
+        if (!poller.isDone()) {
+          await poller.poll({ abortSignal });
+          while (!poller.isDone()) {
+            const delay = sleep(currentPollIntervalInMs, abortSignal);
+            cancelJob = () => abortController.abort();
+            await delay;
+            await poller.poll({ abortSignal });
+          }
+        }
+        switch (state.status) {
+          case "succeeded": {
+            return poller.getResult() as TestGetFile200Response;
+          }
+          case "canceled": {
+            throw new Error("Operation was canceled");
+          }
+          case "failed": {
+            throw state.error;
+          }
+          case "notStarted":
+          case "running": {
+            // Unreachable
+            throw new Error(`polling completed without succeeding or failing`);
+          }
+        }
+      })().finally(() => {
+        resultPromise = undefined;
+      }));
+    },
+
+    onProgress(
+      callback: (state: OperationState<TestGetFile200Response>) => void
+    ): CancelOnProgress {
+      const s = Symbol();
+      progressCallbacks.set(s, callback);
+
+      return () => progressCallbacks.delete(s);
+    },
+
+    isDone(): boolean {
+      return ["succeeded", "failed", "canceled"].includes(state.status);
+    },
+
+    stopPolling(): void {
+      abortController.abort();
+      cancelJob?.();
+    },
+
+    isStopped(): boolean {
+      return resultPromise === undefined;
+    },
+
+    getOperationState(): OperationState<TestGetFile200Response> {
+      return state;
+    },
+
+    getResult(): TestGetFile200Response | undefined {
+      return state.result;
+    },
+
+    toString() {
+      return JSON.stringify({ state });
+    },
+  };
+
+  return poller;
+}
+
+const REJECTED_ERR = new Error("The operation has been aborted");
+
+function sleep(ms: number, signal: AbortSignalLike): Promise<void> {
```

Comment:
This is being duplicated, could it be shared instead?

Reply:
Added common utility.